test/framework/src/main/java/org/elasticsearch/test/InternalAggregationTestCase.java

```diff
@@ -54,9 +54,9 @@ protected T createUnmappedInstance(String name,
 
     public void testReduceRandom() {
         String name = randomAlphaOfLength(5);
-        List<T> inputs = new ArrayList<>();
-        List<InternalAggregation> toReduce = new ArrayList<>();
         int toReduceSize = between(1, 200);
+        List<T> inputs = new ArrayList<>(toReduceSize);
+        List<InternalAggregation> toReduce = new ArrayList<>(toReduceSize);
```

Comment:
Note the extra reduce is added only if we are doing an extra reduce, and toReduce is greater than one.